src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1459,12 +1459,13 @@ int PeerReplayer::do_sync_snaps(const std::string &dir_root) {
     }
     std::chrono::duration<double> duration = clock::now() - start;
     set_last_synced_stat(dir_root, it->first, it->second, duration.count());
-    if (--snaps_per_cycle == 0) {
-      break;
-    }
 
     last_snap_name = it->second;
     last_snap_id = it->first;
+
+    if (--snaps_per_cycle < 1) {
```

Comment:
I don't get it. Please explain.

Reply:
This check seems to be appropriate based on the type of `cephfs_mirror_max_snapshot_sync_per_cycle` and the min value. As `snaps_per_cycle` also decides whether the sync continues, putting it last seems better. These are the reasons I thought while creating it.

Reply:
> This check seems to be appropriate based on the type of `cephfs_mirror_max_snapshot_sync_per_cycle` 

which is uint64_t

and the min value.

min value: 1, which means there will be at least one iteration per sync cycle. I don't see how using `< 1` is fixing anything.

As `snaps_per_cycle` also decides whether the sync continues, putting it last seems better.

Sorry, I don't get how its better. Can you elaborate?

Reply:
Got it, min value is 1. Closing it.